packages/esm-patient-banner-app/src/banner-tags/print-identifier-sticker.component.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button, ModalBody, ModalFooter, ModalHeader } from '@carbon/react';
+import styles from './print-identifier-sticker.scss';
+import { useReactToPrint } from 'react-to-print';
+import { Column } from '@carbon/react';
+import { Grid } from '@carbon/react';
+import { age, displayName, showSnackbar, useConfig } from '@openmrs/esm-framework';
+
+interface PrintIdentifierStickerProps {
+  patient: fhir.Patient;
+  closeModal: () => void;
+}
+
+const PrintIdentifierSticker: React.FC<PrintIdentifierStickerProps> = ({ patient, closeModal }) => {
+  const { t } = useTranslation();
+  const [isPrinting, setIsPrinting] = useState(false);
+  const contentToPrintRef = useRef(null);
+  const [printRetryCount, setprintRetryCount] = useState(0);
+  const { printIdentifierStickerFields, printIdentifierStickerSize, excludePatientIdentifierCodeTypes } = useConfig();
+
+  const headerTitle = t('patientIdentifierSticker', 'Patient Identifier Sticker');
+
+  const onBeforeGetContentResolve = useRef(null);
+
+  useEffect(() => {
+    if (isPrinting && onBeforeGetContentResolve.current) {
+      onBeforeGetContentResolve.current();
+    }
+  }, [isPrinting]);
+
+  const patientDetails = useMemo(() => {
+    const getGender = (gender: string): string => {
+      switch (gender) {
+        case 'male':
+          return t('male', 'Male');
+        case 'female':
+          return t('female', 'Female');
+        case 'other':
+          return t('other', 'Other');
+        case 'unknown':
+          return t('unknown', 'Unknown');
+        default:
+          return gender;
+      }
+    };
+
+    const identifiers =
+      patient?.identifier?.filter(
+        (identifier) => !excludePatientIdentifierCodeTypes?.uuids.includes(identifier.type.coding[0].code),
+      ) ?? [];
+
+    return {
+      id: patient?.id,
+      photo: patient?.photo,
+      name: patient ? displayName(patient) : '',
+      dateOfBirth: patient.birthDate,
+      age: age(patient?.birthDate),
+      gender: getGender(patient?.gender),
+      address: patient?.address,
+      identifiers: identifiers?.length ? identifiers.map(({ value, type }) => ({ value, type })) : [],
```

Comment:
`identifiers` here _always_ has a value. I don't really understand the point of the `map()` call?

```suggestion
      identifiers: identifiers.length > 0 ? identifiers.map(({ value, type }) => ({ value, type })) : [],
```

Reply:
Maybe you wanted:

```ts
identifiers: [...identifiers],
```